miio/aqaracamera.py

```diff
@@ -0,0 +1,291 @@
+    )
+    def ir_on(self):
+        """IR on."""
+        return self.send("set_ir", ["on"])
+
+    @command(
+        default_output=format_output("IR off")
+    )
+    def ir_off(self):
+        """IR off."""
+        return self.send("set_ir", ["off"])
+
+    @command(
+        default_output=format_output("MD on")
+    )
+    def md_on(self):
+        """IR on."""
+        return self.send("set_md", ["on"])
+
+    @command(
+        default_output=format_output("MD off")
+    )
+    def md_off(self):
+        """MD off."""
+        return self.send("set_md", ["off"])
+
+    @command(
+        default_output=format_output("LED on")
+    )
+    def led_on(self):
+        """LED on."""
+        return self.send("set_led", ["on"])
+
+    @command(
+        default_output=format_output("LED off")
+    )
+    def led_off(self):
+        """LED off."""
+        return self.send("set_led", ["off"])
+
+    @command(
+        default_output=format_output("Flip on")
+    )
+    def flip_on(self):
+        """Flip on."""
+        return self.send("set_flip", ["on"])
+
+    @command(
+        default_output=format_output("Flip off")
+    )
+    def flip_off(self):
+        """Flip off."""
+        return self.send("set_flip", ["off"])
+
+    @command(
+        default_output=format_output("Fullstop on")
+    )
+    def fullstop_on(self):
+        """Fullstop on."""
+        return self.send("set_fullstop", ["on"])
+
+    @command(
+        default_output=format_output("Fullstop off")
+    )
+    def fullstop_off(self):
+        """Fullstop off."""
+        return self.send("set_fullstop", ["off"])
+
+    @command(
+        click.argument("time", type=int, default=30),
+        default_output=format_output(
+            "Start pairing for {time} seconds")
+    )
+    def pair(self, timeout: int):
+        """Start (or stop with "0") pairing."""
+        if timeout < 0:
+            raise CameraException("Invalid timeout: %s" % timeout)
+
+        return self.send("start_zigbee_join", [timeout])
+
+    @command()
+    def sd_status(self):
+        """SD card status. TODO: please report output."""
+        return self.send("get_sdstatus")
+
+    @command()
+    def sd_format(self):
+        """TODO: Format SD card? parameters & result unknown."""
+        return self.send("sdformat")
+
+    @command()
+    def arm_status(self):
+        """Return arming information."""
+        # TODO: return a container
+        is_armed = self.send("get_arming")
+        arm_wait_time = self.send("get_arm_wait_time")
+        return {'is_armed': is_armed, 'wait_time': arm_wait_time}
+
+    @command(
+        default_output=format_output("Arming")
+    )
+    def arm(self):
+        """Arm the camera?"""
+        return self.send("set_arming", ["on"])
+
+    @command(
+        default_output=format_output("Disarming")
+    )
+    def disarm(self):
+        """Disarm the camera?"""
+        return self.send("set_arming", ["off"])
```

Comment:
The disarm part is for the gateway.
Disarm also disables the alarm coming from other device like a door sensor if configured to alarm.

This one has a camera so can get the alarm event also from video motion detection but the arm mechanism is the same as the gateways.

Reply:
Ok, I think it can be removed for now. If/when it makes sense to introduce support for the gateway, the camera should derive these from its implementation (I think).